src/Schema.js

```diff
@@ -296,27 +297,37 @@ class Schema {
     }
   }
 
-  // getForState(item){} // least important one, not even sure if needed
-
-  /**
-   * gets all data for the transporter except for the primary key
-   * (an item can't have a key until its created in the transporter for example )
-   * has to wait until all foreing keys are created. (this is tricky,
-   * pls aks johannes before implementing).
-   *
-   * initialData is an object with data, that has to be in the result as well
-   */
-  // getFor(source: DataSource, item: Object, initialData: Object) {
-  //
-  // }
-
-  /**
-   * gets the transporter primary key. The function is only called when the key
-   * already exists, so this should be sync. returns { <key_name> : <key_value> }
-   */
-  // getPrimaryKey(source: DataSource, item: Object) {
-  //
-  // }
+  getFor(source: DataSource, item: Item, initialData: Object = {}) {
+    const prefix = source === TARGET.CLIENT_STORAGE ? '_' : '';
+    const returnData = _.cloneDeep(initialData);
+    return item.onceReadyFor(source).then(() => {
```

Comment:
Ah ok, so this means, the actual getFor() is always synchronous and it is wrapped by an async check for the relational items to be ready.

This isn't that hard. Will solve this until the end of the week :)